pkg/reconciler/namespace/namespace.go

```diff
@@ -174,6 +178,27 @@ func (r *Reconciler) reconcileServiceAccountAndRoleBindings(ctx context.Context,
 		return fmt.Errorf("track role binding '%s': %v", rb.Name, err)
 	}
 
+	if sa.Name == resources.IngressServiceAccountName || sa.Name == resources.FilterServiceAccountName {
+		// check for existence of brokerPullSecret, and skip copy if it already exists
+		secretFound := false
+		for _, v := range sa.ImagePullSecrets {
+			if fmt.Sprintf("%s", v) == ("{" + r.brokerPullSecretName + "}") {
+				secretFound = true
+				break
+			}
+		}
+		if (!secretFound) {
```

Comment:
If you do what I suggest above, you can get rid of this if clause altogether since it will only be executed if the secret is not found (by virtue of returning when found).